framework/decode/custom_vulkan_ascii_consumer.h

```diff
@@ -80,18 +80,66 @@ inline std::string EnumPointerDecoderToString(PointerDecoder<VkEnumType>* pObj)
     return pDecodedObj ? ('"' + util::ToString(*pDecodedObj) + '"') : "null";
 }
 
+/// @note This never did anything useful but remains to trigger the assert if
+/// new types pass through here which need a specialisation.
+/// @note There are structures that have data fields that can be interpreted
+/// correctly...if we want to output interpreted void pointer data we can
+/// implement custom handlers for those structures.
 template <typename T>
 inline std::string DataPointerDecoderToString(const T& pObj)
 {
-    // NOTE : Currently we just output the integer value of data void pointers, for
-    //  fields that use a PointerDecoder<> we output the address of the decoder
-    //  itself...this isn't terribly useful, but the capture time address wouldn't
-    //  be a terribly useful thing anyway.  There are structures that have data
-    //  fields that can be interpreted correctly...if we want to output interpreted
-    //  void pointer data we can implement custom handlers for those structures.
+    assert(0 == "We should never be calling this. Investigate up the call stack to see whether type T needs support.");
     return "\"" + util::PtrToString(pObj) + "\"";
 }
 
+/// Convert pointers to unknown types coming through as uint64_t.
+/// ValueDecoder::DecodeAddress() is turning these pointers into uint64_ts
+/// in code generated by `base_decoder_body_generator.py`
+/// These hold:
+/// `const struct AHardwareBuffer*` from
+/// `vkGetAndroidHardwareBufferPropertiesANDROID`, 'const void *' from
+/// `vkGetMemoryHostPointerPropertiesEXT`, and from
+/// `vkCmdSetCheckpointNV`, `IDirectFB*` from
+/// `vkGetPhysicalDeviceDirectFBPresentationSupportEXT`, `struct _screen_window*` from
+/// `kGetPhysicalDeviceScreenPresentationSupportQNX`, `Display*` from
+/// `vkGetPhysicalDeviceXlibPresentationSupportKHR`, and from
+/// `vkAcquireXlibDisplayEXT`, and from
+/// `vkGetRandROutputDisplayEXT`, `struct wl_display*` from
+/// `vkGetPhysicalDeviceWaylandPresentationSupportKHR`, `xcb_connection_t*` from
+/// `vkGetPhysicalDeviceXcbPresentationSupportKHR`, Win32 memory `HANDLE` from
+/// `vkGetMemoryWin32HandlePropertiesKHR`.
+/// The uint64_t is treated as an address in each case.
+inline std::string DataPointerDecoderToString(const uint64_t buffer)
+{
+    std::string str{ "\"" + util::PtrToString(buffer) + "\"" };
+    return str;
+}
+
+/// Convert pointers to arrays of void pointers to a string of the captured address.
+template <typename T, typename U>
+inline std::string DataPointerDecoderToString(PointerDecoder<T, U>* pDecoder)
+{
+    if (nullptr == pDecoder)
```

Comment:
I suggest just a GFXRECON_ASSERT() here and don't bother returning a NULL.

Reply:
Done